### ISSUE TITLE
Can not extend an extensible variant which is an alias

### ISSUE THREAD

**Issue by craff** [ocaml/ocaml#12353]
This code is rejected:

```
module A = struct
  type t = ..
end

module B = struct
  type t = A.t
end

module C = struct
  type B.t += X
end
```

I was hit by the above example when trying to give a different presentation to the end user than the original implementation.
I think this code should be accepted.

**Comment by lthls** [comment]
You have to re-export the fact that it is an open type:
```ocaml
module B = struct
  type t = A.t = ..
end
```

**Comment by craff** [comment]
Argh, I missed that. In fact even the following is working (which is what I really need) :+1: 
```

module A = struct
  type t = ..
  type t += A
end

module B = struct
  type t = A.t = ..
  type t += A = A.A
end

module C = struct
  type B.t += X
end
```

But I do not see this in the documentation (chapter 12 section 14), except by reading the BNF, but this is almost useless if you don't know what it is for. 

By the way, I think 
```
type a = A | ..
```

should be supported, as a short cut.

**Comment by craff** [comment]
fixed by PR  #12359 on the doc.
